### PR TITLE
Add Robot 101 post on The Code

### PR DESCRIPTION
**Pull request by sedders123**

### CODE REVIEW

**Comment by antoinepetty** [comment]
Really good, thanks James! I think it'd be worth adding a reference to the docs and maybe some Python sources for where to find more concrete info

File: _posts/2023-06-29-101-code.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+your work from these unpredictable events, ensure you regularly back up your
+code. Better yet, use a Version Control System (VCS) like Git. A VCS not only
+provides a backup of your code but also tracks changes, making it easy to
+identify when and where things might have gone wrong or right!
+
+## The Six T's: Test, Test, Test, Test, Test, Test!
+
+If there's one golden rule in the world of coding, it's this: test your code.
+Then, test it again, and keep testing! You can utilise our web interface to help
+you quickly iterate on your robot, by live streaming the logs and restarting
```
Comment (WillB97) [inline _posts/2023-06-29-101-code.md:37]: Should we also mention webcam images as well as logs?

File: _posts/2023-06-29-101-code.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+execution of your code. The more robust your testing, the more reliable your
+robot will be when it counts.
```
Comment (sedders123) [inline _posts/2023-06-29-101-code.md:39]:
```suggestion
you quickly iterate on your robot, by live streaming the logs and viewing
the last captured image. You can also use the web interface to stop and 
restart the execution of your code. The more robust your testing, the 
more reliable your robot will be when it counts.
```

File: _posts/2023-06-29-101-code.md (hunk 1 above)
Comment (sedders123) [reply _posts/2023-06-29-101-code.md]: [f5c2794](https://github.com/srobo/website/pull/510/commits/f5c2794060f7aec8adbcc26ea8cd580ecc65d091)

File: _posts/2023-06-29-101-code.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+acting. It's essential to have a solid control loop in your code where your
+robot senses its environment and then decides what to do next based on that
+information. This loop should be at the core of your robot's operation, allowing
+it to adapt and react to its surroundings.
+
+## Backup Your Code: The Magic of Version Control Systems
+
+Mistakes and accidents happen, files get lost, and computers crash. To protect
+your work from these unpredictable events, ensure you regularly back up your
+code. Better yet, use a Version Control System (VCS) like Git. A VCS not only
```
Comment (samjmartin04) [inline _posts/2023-06-29-101-code.md:29]: Is a specific mention of github itself worth it? I'd only heard of github and not git specifically before SR

**Review by samjmartin04 — APPROVED** [review]

File: _posts/2023-06-29-101-code.md (hunk 3 above)
Comment (sedders123) [reply _posts/2023-06-29-101-code.md:29]:
```suggestion
code. Better yet, use a Version Control System (VCS) like Git with GitHub. A VCS not only
```

Comment (sedders123) [reply _posts/2023-06-29-101-code.md:29]:
```suggestion
code. Better yet, use a Version Control System (VCS) like Git with GitHub. A VCS not only
```

File: _posts/2023-06-29-101-code.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+identify when and where things might have gone wrong or right!
+
+## The Six T's: Test, Test, Test, Test, Test, Test!
+
+If there's one golden rule in the world of coding, it's this: test your code.
+Then, test it again, and keep testing! You can utilise our web interface to help
+you quickly iterate on your robot, by live streaming the logs and viewing
+the last captured image. You can also use the web interface to stop and 
+restart the execution of your code. The more robust your testing, the 
+more reliable your robot will be when it counts.
```
Comment (sedders123) [inline _posts/2023-06-29-101-code.md:40]:
```suggestion
more reliable your robot will be when it counts!
```

File: _posts/2023-06-29-101-code.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+## The Benefits of Friendly Competition
+
+If you have a large enough team, consider developing competing implementations
+of certain features or systems. This can be an effective way to explore
+different solutions and spark innovation. You can then choose the best elements
+from each, or select the most efficient solution overall.
+
+## Embrace Imperfections: Sensor Noise
+
+It's crucial to remember that real-world data are often noisy and imperfect.
```
Comment (sedders123) [inline _posts/2023-06-29-101-code.md:51]:
```suggestion
It's crucial to remember that real-world data is often noisy and imperfect.
```

File: _posts/2023-06-29-101-code.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+Your sensors may pick up interference, or there could be slight variations in
+readings. Your code should be robust enough to handle this noise and still make
+effective decisions. Techniques such as sensor fusion or filtering can be highly
+beneficial here.
```
Comment (sedders123) [inline _posts/2023-06-29-101-code.md:55]:
```suggestion
effective decisions. Techniques such as sensor fusion or filtering (like 
taking an average of multiple readings) can be highly beneficial here.
```

File: _posts/2023-06-29-101-code.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+## The Power of Documentation
+
+Our [Docs](https://studentrobotics.org/docs) are a goldmine of knowledge and 
+should be your first port of call whenever you're unsure about something. Much 
+like a good textbook or reference guide, our Docs are designed to support you in
+understanding your robot better. From getting started with your robot kit, to 
+understanding the APIs, to troubleshooting common issues, the Docs cover a broad
+spectrum of topics you may encounter.
```
Comment (sedders123) [inline _posts/2023-06-29-101-code.md:64]:
```suggestion
understanding your robot better. From getting started with your robot kit, to 
understanding the APIs, to troubleshooting common issues; the Docs cover a broad
spectrum of topics you may encounter.
```

File: _posts/2023-06-29-101-code.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+like a good textbook or reference guide, our Docs are designed to support you in
+understanding your robot better. From getting started with your robot kit, to 
+understanding the APIs, to troubleshooting common issues; the Docs cover a broad
+spectrum of topics you may encounter.
+
+<br/>
+
+Coding your robot can be a challenging yet rewarding experience. It's where the
+magic happens, where the inert pieces of metal and circuits come to life and
+engage with the world. We hope these tips and strategies will help you avoid
```
Comment (sedders123) [inline _posts/2023-06-29-101-code.md:70]:
```suggestion
Coding your robot can be a challenging yet rewarding experience. It's where the
magic happens, where the inert pieces of metal, wood, and circuitry come to life
and engage with the world. We hope these tips and strategies will help you avoid
```

**Review by WillB97 — APPROVED** [review]

**Pull request merged**